### PR TITLE
Round up Intersects implementations

### PR DESCRIPTION
**Pull request by rmanoka**
This PR completes the `Intersects` implementation for all pairs of current geo-types (discounting possible bugs)!

+ Add OGC-SA descriptions to geo-types docs
+ Verify existing implementations
+ Add implementations for containers: `Geometry`, `GeometryCollection`

See [here](//georust.github.io/geo/support/intersects.html) for table describing the implementation status / map.

Interestingly, we didn't need geom-graph for this trait (presumably because this trait doesn't distinguish between interior and boundary).  I also removed any usage of `Contains` trait in this trait for easier verification of correctness.

### CODE REVIEW

File: geo/src/algorithm/intersects/point.rs
Comment (michaelkirk) [inline geo/src/algorithm/intersects/point.rs:8]: 👍

**Review by michaelkirk — COMMENTED** [review]
One correctness concern around the triangle semantics comment, everything else was just style nits which you can feel free to ignore.

**Review by michaelkirk — APPROVED** [review]
Thanks for kicking this off @rmanoka!

File: geo-types/src/coordinate.rs
Comment (rmanoka) [reply geo-types/src/coordinate.rs:22]: I left this intentionally vague, as I also wasn't sure of the exact requirements.  For instance, inf - inf is nan, so any operation that computes diffs (would end up producing nan coordinates if starting with inf).

File: geo-types/src/lib.rs
Comment (rmanoka) [reply geo-types/src/lib.rs:12]: Well, we just "aim to"; nothing wrong in aiming high!

File: geo-types/src/point.rs
Comment (rmanoka) [reply geo-types/src/point.rs:16]: I'm referring to validity requirement of Coordinate (which is vague too :) )

File: geo-types/src/triangle.rs
Comment (rmanoka) [reply geo-types/src/triangle.rs:6]: Yeah, definitely "not collinear".  Distinctness follows from it.

**Comment by rmanoka** [comment]
Seem to be getting a few CI failures due to some error in compiling proj: https://github.com/rmanoka/geo/runs/1213075197 .  Any idea why?

```
--- stderr
pkg-config unable to find existing libproj installation: Failed to run `"pkg-config" "--libs" "--cflags" "proj" "proj >= 7.1.0"`: No such file or directory (os error 2)
building libproj from source
thread 'main' panicked at '
failed to execute command: No such file or directory (os error 2)
is `cmake` not installed?

build script failed, must exit now', /github/home/.cargo/registry/src/github.com-1ecc6299db9ec823/cmake-0.1.44/src/lib.rs:885:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.

Error: Process completed with exit code 101.
```

**Comment by michaelkirk** [comment]
> Seem to be getting a few CI failures due to some error in compiling proj: https://github.com/rmanoka/geo/runs/1213075197 . Any idea why?

We've made a fair number of changes to proj compilation and the underlying ci docker images... I suspect I broke something. I'll take a look!

**Comment by michaelkirk** [comment]
>> Seem to be getting a few CI failures due to some error in compiling proj: https://github.com/rmanoka/geo/runs/1213075197 . Any idea why?
>
> We've made a fair number of changes to proj compilation and the underlying ci docker images... I suspect I broke something. I'll take a look!

Ok, I've fixed CI in #517

**Comment by michaelkirk** [comment]
bors try

This should be good to merge now - but I’m not sure if you were wanting to make any further changes, so I’ll leave it to you to merge @rmanoka

**Comment by rmanoka** [comment]
@michaelkirk Thanks for fixing the CI!  I'm merging this now

bors r+

**Pull request merged**
